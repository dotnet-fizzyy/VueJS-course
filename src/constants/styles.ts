export const FontSizeDirectiveName: string = 'font-size';
export const FontSizeDefaultValue: number = 14;
