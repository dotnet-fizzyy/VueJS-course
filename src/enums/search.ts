export enum SearchByOptionNames {
    Title = 'title',
    Genre = 'genres',
}

export enum SortByOptionsNames {
    ReleaseDate = 'release_date',
    Rating = 'rating',
}

export enum SearchQueryParams {
    SearchTerm = 'term',
    SearchByOption = 'searchBy',
    SortByOption = 'sortBy',
}
