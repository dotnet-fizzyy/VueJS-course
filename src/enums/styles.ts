export enum FontStyle {
    Normal = 'normal',
    Bold = 'bold',
    Italic = 'italic',
}

export enum FontSize {
    Small = 'small',
    Medium = 'medium',
    Large = 'large',
}
