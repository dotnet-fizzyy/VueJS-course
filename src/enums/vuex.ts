export enum FilmGetterKeys {
    FilmsPreviews = 'filmsPreviews',
    AvailableCount = 'availableCount',
    GetFilmFullDescriptionById = 'getFilmFullDescriptionById',
}
