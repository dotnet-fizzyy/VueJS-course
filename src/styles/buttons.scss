@import './variables';

.primary-button {
  width: 100%;
  height: 100%;
  color: #fff;
  background-color: $dark-pink;
  border-radius: 4px;
  border: 1px solid $dark-pink;
  cursor: pointer;
}
