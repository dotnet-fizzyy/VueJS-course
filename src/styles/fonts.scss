.normal {
  font-weight: normal;
}

.bold {
  font-weight: bold;
}

.italic {
  font-style: italic;
}
