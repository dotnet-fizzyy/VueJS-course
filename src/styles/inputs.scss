@import "variables";

.input {
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  border-radius: 5px;
  border: none;
  outline: none;
  color: $white;
  background: getRgbaGrey(0.7);
  padding: 0 10px;
}

