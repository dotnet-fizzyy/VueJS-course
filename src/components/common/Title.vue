<template>
    <span :class="style"><slot /></span>
</template>

<script lang="ts">
import Vue from 'vue';
import { FontSize, FontStyle } from '@/enums/styles';

export interface TitleProps {
    fontStyle?: FontStyle;
    fontSize?: FontSize;
    title: string;
}

export default Vue.extend({
    name: 'Title',
    props: {
        fontStyle: {
            type: String,
            default: FontStyle.Normal,
        },
        fontSize: {
            type: String,
            default: FontSize.Medium,
        },
    },
    computed: {
        style(): string[] {
            const fontStyle: string = this.$style[this.fontStyle];
            const fontSize: string = this.$style[this.fontSize];

            return [this.$style.font, fontStyle, fontSize];
        },
    },
});
</script>

<style lang="scss" scoped module>
@import '../../styles/fonts';
</style>
