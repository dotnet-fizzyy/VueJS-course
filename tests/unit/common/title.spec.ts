import Title from '@/components/common/AppTitle.vue';
import { FontStyle } from '@/enums/styles';
import { shallowMount } from '@vue/test-utils';

describe('AppTitle.vue Tests', () => {
    test('Should render', () => {
        //Arrange
        const wrapper = shallowMount(Title, {
            slots: {
                default: 'Hello world!',
            },
            propsData: {
                fontSize: 20,
                fontStyle: FontStyle.Italic,
            },
        });

        //Act & Assert
        expect(wrapper.element).toMatchSnapshot();
    });

    test('Should contain correct text message', () => {
        //Arrange
        const message = 'Some message';

        const wrapper = shallowMount(Title, {
            slots: {
                default: message,
            },
            propsData: {
                fontSize: 20,
                fontStyle: FontStyle.Italic,
            },
        });

        //Act & Assert
        expect(wrapper.text()).toEqual(message);
    });
});
