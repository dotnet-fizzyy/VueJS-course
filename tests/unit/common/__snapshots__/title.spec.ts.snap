// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTitle.vue Tests Should render 1`] = `
<span
  class="font italic"
  style="font-size: 20px;"
>
  Hello world!
</span>
`;
